{
  "url": "https://api.github.com/repos/facebook/react/issues/16165",
  "repository_url": "https://api.github.com/repos/facebook/react",
  "labels_url": "https://api.github.com/repos/facebook/react/issues/16165/labels{/name}",
  "comments_url": "https://api.github.com/repos/facebook/react/issues/16165/comments",
  "events_url": "https://api.github.com/repos/facebook/react/issues/16165/events",
  "html_url": "https://github.com/facebook/react/issues/16165",
  "id": 470604166,
  "node_id": "MDU6SXNzdWU0NzA2MDQxNjY=",
  "number": 16165,
  "title": "Unexpected behavior with ref callback and focus()",
  "user": {
    "login": "ZYinMD",
    "id": 32368482,
    "node_id": "MDQ6VXNlcjMyMzY4NDgy",
    "avatar_url": "https://avatars3.githubusercontent.com/u/32368482?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/ZYinMD",
    "html_url": "https://github.com/ZYinMD",
    "followers_url": "https://api.github.com/users/ZYinMD/followers",
    "following_url": "https://api.github.com/users/ZYinMD/following{/other_user}",
    "gists_url": "https://api.github.com/users/ZYinMD/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/ZYinMD/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/ZYinMD/subscriptions",
    "organizations_url": "https://api.github.com/users/ZYinMD/orgs",
    "repos_url": "https://api.github.com/users/ZYinMD/repos",
    "events_url": "https://api.github.com/users/ZYinMD/events{/privacy}",
    "received_events_url": "https://api.github.com/users/ZYinMD/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "closed",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 1,
  "created_at": "2019-07-19T23:23:28Z",
  "updated_at": "2019-07-20T00:00:43Z",
  "closed_at": "2019-07-20T00:00:42Z",
  "author_association": "NONE",
  "body": "**Do you want to request a *feature* or report a *bug*?**\r\nbug (I know it's more likely a bug of my own code, but I can't find out why)\r\n\r\n**What is the current behavior?**\r\n### [CodeSandbox Demo](https://codesandbox.io/s/quizzical-gagarin-bzvf1)\r\nIt's really hard to describe my issue in English, the only way to understand is looking at the demo. Basically I need to programmatically focus on any element in a dynamic array by the element id, so I created an object to store all the refs by id, but the reference to the object sometimes doesn't update as expected.\r\n\r\n**What is the expected behavior?**\r\nClicking the last button should be able to create then focus on input5.\r\n\r\n**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**\r\nv16.8.6, v3.0.1, Chrome, Win10, CodeSandbox. I don't know if previous versions or React is the same.\r\n",
  "closed_by": {
    "login": "ZYinMD",
    "id": 32368482,
    "node_id": "MDQ6VXNlcjMyMzY4NDgy",
    "avatar_url": "https://avatars3.githubusercontent.com/u/32368482?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/ZYinMD",
    "html_url": "https://github.com/ZYinMD",
    "followers_url": "https://api.github.com/users/ZYinMD/followers",
    "following_url": "https://api.github.com/users/ZYinMD/following{/other_user}",
    "gists_url": "https://api.github.com/users/ZYinMD/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/ZYinMD/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/ZYinMD/subscriptions",
    "organizations_url": "https://api.github.com/users/ZYinMD/orgs",
    "repos_url": "https://api.github.com/users/ZYinMD/repos",
    "events_url": "https://api.github.com/users/ZYinMD/events{/privacy}",
    "received_events_url": "https://api.github.com/users/ZYinMD/received_events",
    "type": "User",
    "site_admin": false
  }
}
